src/test/kotlin/org/rust/lang/core/resolve/RustTypeAwareResolveTestCase.kt

```diff
@@ -25,9 +25,12 @@ class RustTypeAwareResolveTestCase : RustResolveTestCaseBase() {
 
     fun testHiddenInherentImpl() = checkIsBound()
     fun testWrongInherentImpl() = checkIsUnbound()
-    fun testNonInherentImpl() = checkIsBound()
     fun testImplGenericsStripped() = checkIsBound()
 
+    fun testNonInherentImpl1() = checkIsBound()
+    fun testNonInherentImpl2() = checkIsBound()
+    fun testNonInherentImpl3() = checkIsBound()
```

Comment:
`Impl2` and `Impl3` cases should resolve to the same method:

http://play.integer32.com/?gist=8eac4db6da8c8198687bda55139f8a36